### PR TITLE
Alps 661 load scene time

### PR DESCRIPTION
**Pull request by rroux-gpsw**
Refactor of media location. It is now owned by the scene, not the renderer.
Add a sync setting on scene.

### CODE REVIEW

File: src/render/RenderManager.js [hunk 1]
```diff
@@ -667,15 +590,16 @@ FORGE.RenderManager.prototype._setBackgroundRenderer = function(type)
     this._clearBackgroundRenderer();
 
     var config = {};
-    var mediaConfig = /** @type {SceneMediaConfig} */ (this._sceneConfig.media);
+    var media = this._viewer.story.scene.media;
+    var mediaConfig = /** @type {SceneMediaConfig} */ (media.config);
```
Comment (aberthet-gpsw) [inline src/render/RenderManager.js]: Cast is not necessary anymore.

File: src/render/RenderManager.js [hunk 2]
```diff
@@ -775,14 +702,24 @@ FORGE.RenderManager.prototype._setBackgroundRenderer = function(type)
...
+
     if (vrEnabled === true)
     {
         this.log("VR on - background type = MESH");
         this._backgroundRendererType = FORGE.BackgroundType.MESH;
         return;
     }
 
-    var mediaConfig = /** @type {SceneMediaConfig} */ (this._sceneConfig.media);
+    var mediaConfig = /** @type {SceneMediaConfig} */ (media.config);
```
Comment (aberthet-gpsw) [inline src/render/RenderManager.js]: Cast is not necessary anymore.

File: src/story/Story.js [hunk 3]
```diff
@@ -434,128 +345,102 @@ FORGE.Story.prototype._createScenes = function(config)
...
     //The scene has no group so nullify the _groupUid
-    if(scene.hasGroups() === false)
+    if(nextScene.hasGroups() === false)
     {
         this._groupUid = null;
 
         if(this._onGroupChange !== null)
         {
-            this._onGroupChange.dispatch();
+            this._onGroupChange.dispatch({ groupUid: this._groupUid });
```
Comment (aberthet-gpsw) [inline src/story/Story.js]: `groupUid` is being minified, you need to cast the object.

File: src/story/Story.js [hunk 4]
```diff
@@ -434,128 +345,102 @@ FORGE.Story.prototype._createScenes = function(config)
...
         }
+    }
+    else if (nextScene.hasGroups() === true && nextScene.hasGroup(this._groupUid) === false)
+    {
+        this._setGroupUid(nextScene.groups[0].uid);
+    }
 
+    if(this._onSceneLoadRequest !== null)
+    {
+        this._onSceneLoadRequest.dispatch({ sceneUid: this._sceneUid });
```
Comment (aberthet-gpsw) [inline src/story/Story.js]: Same here.

File: src/story/Story.js [hunk 5]
```diff
@@ -434,128 +345,102 @@ FORGE.Story.prototype._createScenes = function(config)
...
+ * @private
+ */
+FORGE.Story.prototype._sceneLoadStartHandler = function()
+{
+    this.log("scene load start");
 
     if(this._onSceneLoadStart !== null)
     {
-        this._onSceneLoadStart.dispatch({ uid: scene.uid });
+        this._onSceneLoadStart.dispatch({ sceneUid: this._sceneUid });
```
Comment (aberthet-gpsw) [inline src/story/Story.js]: Same here. You also forgot to update `FORGE.PluginManager#_sceneLoadStartHandler` which is still using `did`.

File: src/story/Story.js [hunk 6]
```diff
@@ -566,14 +451,16 @@ FORGE.Story.prototype._sceneLoadStart = function(event)
...
  */
-FORGE.Story.prototype._sceneLoadComplete = function()
+FORGE.Story.prototype._sceneLoadCompleteHandler = function()
 {
+    this.log("scene load complete");
+
     if(this._onSceneLoadComplete !== null)
     {
-        this._onSceneLoadComplete.dispatch();
+        this._onSceneLoadComplete.dispatch({ sceneUid: this._sceneUid });
```
Comment (aberthet-gpsw) [inline src/story/Story.js]: Same here.

File: src/story/Story.js [hunk 7]
```diff
@@ -612,7 +499,7 @@ FORGE.Story.prototype._setGroupUid = function(uid)
 
         if(this._onGroupChange !== null)
         {
-            this._onGroupChange.dispatch();
+            this._onGroupChange.dispatch({ groupUid: this._groupUid });
```
Comment (aberthet-gpsw) [inline src/story/Story.js]: Same here.

**Comment by sw-team-release-gpsw** [comment]
[CLA assistant check](https://cla.gopro.com/sign) <br/>All committers have signed the CLA.

**Pull request merged**